Feature/fltr/prepare for use shared prefs

manualcheg opened the pull request:
1. Добавлены модули зависимостей всех слоев для koin
2. SharedPreferences описана как зависимость koin
3. Создан класс- модель данных Filter и сопутствующие для хранения значений фильтров в SharedPreferences.
4. Создан класс для хранения констант экрана фильтров
5. Сделан класс конвертер для преобразования Filter в параметры поискового запроса
6. Начато создание реализации методов FilterOptionsDataSourceImpl

1. Добавлен класс между SharedPrefs и DataSource
2. создан новый класс для хранения объекта Filter до применения фильтров к поиску и методы для добавления параметров фильтрации

1. В интерфейс для работы с параметрами фильтрации добавлен метод очистки
2. Доделана реализация интерфейса для работы с параметрами фильтрации
3. Реализованы методы добавления значений параметров фильтрации

manualcheg committed:
1. Добавлены модули зависимостей всех слоев для koin
2. SharedPreferences описана как зависимость koin
3. Создан класс- модель данных Filter и сопутствующие для хранения значений фильтров в SharedPreferences.
4. Создан класс для хранения констант экрана фильтров
5. Сделан класс конвертер для преобразования Filter в параметры поискового запроса
6. Начато создание реализации методов FilterOptionsDataSourceImpl

manualcheg committed:
1. Добавлен класс между SharedPrefs и DataSource
2. создан новый класс для хранения объекта Filter до применения фильтров к поиску и методы для добавления параметров фильтрации

manualcheg committed:
1. В интерфейс для работы с параметрами фильтрации добавлен метод очистки
2. Доделана реализация интерфейса для работы с параметрами фильтрации
3. Реализованы методы добавления значений параметров фильтрации

manualcheg committed:
1. Созданы в классе FilterOptionsDataSource функции добавления значений фильтров

manualcheg committed:
1. Созданы в классе LocalCache функции добавления значений фильтров

manualcheg committed:
Merge remote-tracking branch 'origin/feature/FLTR/prepare_for_use_SharedPrefs' into feature/FLTR/prepare_for_use_SharedPrefs

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/filter/data/dataSourceImpl/FilterOptionsDataSourceImpl.kt:
```diff
@@ -15,13 +18,32 @@ class FilterOptionsDataSourceImpl(
     }
 
     override fun putFilterOptions(options: Filter) {
         filterDb.putFilterOptions(options)
```
Здесь, логично будет очищать localCache

JudjinGM commented on app/src/main/java/ru/practicum/android/diploma/filter/data/db/FilterDbImpl.kt:
```diff
@@ -2,16 +2,24 @@ package ru.practicum.android.diploma.filter.data.db
-        return Gson().fromJson(filterFromSharedPrefs, Filter::class.java)
+class FilterDbImpl(private val sharedPrefs: SharedPreferences) : FilterDb {
+    override fun getFilterOptions(): Filter? {
+        val options = sharedPrefs.getString(FILTER, "")
```
"" можно заменить на константу DEFAULT_VALUE

JudjinGM reviewed (commented):
ревью

manualcheg committed:
1. Переименованы классы
2. Добавлена очистка FilterLocalCache после полу

JudjinGM committed:
1. Перенес модели в common.
2. Добавил очистку кеша после сохранения фильтров d sharedPref.
3. Внес изменения в SarchRepository, для того чтобы проект собирался
